### DISCUSSION TITLE
Use of custom make in eclipse IDE for C project

### DISCUSSION THREAD

**Discussion by soumyaranjansabat** [eclipse-embed-cdt/eclipse-plugins#507] [Managed Build Plug-ins Q&A]
I already have a makefile project in C. There are multiple makefiles that are used through the command line to help automate the build process as multiple folders are available in the project.

How can the same makefiles be used in my Eclipse project for building / compiling the code?

**Comment by ilg-ul** [comment]
The Managed Build Plug-ins focus on managed projects, in other words projects whose build processes are managed by Eclipse, not by the user with manually crafted build configurations, like makefiles.

Although it is possible to use Eclipse unmanaged projects, it is a bit tricky to configure the indexer to reflect the actual build configuration defined in the make files.

I personally avoided this, so I advise you to google for possible solutions available for Eclipse CDT, and try to apply them to Embedded CDT.

**Comment by TommyMurphyTM1234** [comment]
If you really want to use a Makefile project then this may help. It doesn't deal specifically with the Embedded CDT plugins but the general principles should be very similar.

https://mcuoneclipse.com/2017/07/22/tutorial-makefile-projects-with-eclipse/
